# Portfolio Automotive Sales Data with MySQL

# Introduction

Understanding and harnessing the power of data is vital for making informed decisions that can propel a business toward success. In this portfolio, I will delve into the intricacies of automotive order sales data. The dataset at hand encapsulates a wealth of information about the company's transactions, offering valuable insights that can be leveraged to enhance business strategies and drive growth.

# Data Overview

The dataset comprises various attributes that shed light on the dynamics of the automotive sales. Each row in the dataset represents a unique order, and the columns provide details such as order number, quantity ordered, price per item, order line number, total sales, order date, order status, product line, manufacturer's suggested retail price (MSRP), product code, customer name, city, country, and deal size.

# Key Questions for the Analysis

As we navigate through the data, I aim to address pertinent questions that hold the potential to guide strategic decision-making for the supermarket business. Here are four questions that will be explored in this analysis:

1. What are the top-selling product lines, and how can the supermarket optimize inventory based on these insights?

2. How does the sales performance vary across different cities and countries? Are there geographical patterns that can inform marketing and expansion strategies?

3. Can we identify trends or seasonal patterns in sales over time, and how can this information be utilized for effective promotional campaigns and resource allocation?

4. What is the correlation between the variance of Manufacturer's Suggested Retail Price (MSRP) margin with actual price sold and sales, and how does this relationship impact the overall pricing strategy and market positioning of the supermarket?

# 1. Understanding the Product Lines

This query is to calculates the total quantity from the `automotive_order` table of ordered products for each unique `productline`, grouping the results accordingly. The query then presents the `productline` along with their respective total quantities sold as `total_quantity_sold`, ordered in descending order by the highest `total_quantity_sold`.

In [112]:
%%sql

SELECT
    `productline`,
    SUM(`quantityordered`) AS `total_quantity_sold`,
    CONCAT('$',FORMAT (SUM(sales),2)) AS `total_sales_in_usd)`
FROM 
    automotive_co.automotive_order
GROUP BY
    `productline`
ORDER BY 
    `total_quantity_sold` DESC
;


7 rows affected.


[('Classic Cars', Decimal('33992'), '$3,919,615.66'),
 ('Vintage Cars', Decimal('21069'), '$1,903,150.84'),
 ('Motorcycles', Decimal('11663'), '$1,166,388.34'),
 ('Trucks and Buses', Decimal('10777'), '$1,127,789.84'),
 ('Planes', Decimal('10727'), '$975,003.57'),
 ('Ships', Decimal('8127'), '$714,437.13'),
 ('Trains', Decimal('2712'), '$226,243.47')]

After reviewing the table, it's evident that `Classic Cars` and `Vintage Cars` collectively outsell all other product lines combined. This insight suggests an opportunity to strategically focus production efforts and optimize inventory settings. Adjusting resources to accommodate higher storage capacity for Classic Cars and Vintage Cars while minimizing emphasis on less desired products can enhance efficiency and meet market demand more effectively.

The company should consider implementing a `targeted marketing strategy for Classic Cars and Vintage Cars`. Highlight their unique features and historical appeal to `increase awareness and drive higher sales`. Consider exclusive promotions to further incentivize purchases and capitalize on their popularity.

# 2. Looking at the Numbers Geographically



In [34]:
%%sql

SELECT
    `country`,
    SUM(`quantityordered`) AS `total_quantity_sold`,
    CONCAT('$',FORMAT(SUM(SALES),2)) AS `total_sales_in_usd`
FROM
    automotive_co.automotive_order
GROUP BY
    country
ORDER BY 
    `total_quantity_sold` DESC
;

19 rows affected.


[('USA', Decimal('35659'), '$3,627,982.83'),
 ('Spain', Decimal('12429'), '$1,215,686.92'),
 ('France', Decimal('11090'), '$1,110,916.52'),
 ('Australia', Decimal('6246'), '$630,623.10'),
 ('UK', Decimal('5013'), '$478,880.46'),
 ('Italy', Decimal('3773'), '$374,674.31'),
 ('Finland', Decimal('3192'), '$329,581.91'),
 ('Norway', Decimal('2842'), '$307,463.70'),
 ('Singapore', Decimal('2760'), '$288,488.41'),
 ('Canada', Decimal('2293'), '$224,078.56'),
 ('Denmark', Decimal('2197'), '$245,637.15'),
 ('Germany', Decimal('2148'), '$220,472.09'),
 ('Sweden', Decimal('2006'), '$210,014.21'),
 ('Austria', Decimal('1974'), '$202,062.53'),
 ('Japan', Decimal('1842'), '$188,167.81'),
 ('Switzerland', Decimal('1078'), '$117,713.56'),
 ('Belgium', Decimal('1074'), '$108,412.62'),
 ('Philippines', Decimal('961'), '$94,015.73'),
 ('Ireland', Decimal('490'), '$57,756.43')]

Leveraging the considerable success in the US market, the company is poised to strengthen its foothold through strategic expansion initiatives. This involves investing in market penetration efforts, exploring potential partnerships, and considering the establishment of new outlets. Concurrently, the focus should extend to operational optimization, embracing lean processes, negotiating advantageous supplier agreements, and streamlining logistics for heightened efficiency and reduced costs. In tandem, the company must proactively address the risks associated with overreliance on a single market. This includes diversifying market exposure, implementing robust risk management strategies, and maintaining a vigilant stance towards evolving market conditions.







- - To maximize resource efficiency, let's delve into US market specifics by adding the `city` and `dealsize` columns to our table.

In [63]:
%%sql

/* LIMIT function used for data clarity */

SELECT
    `city`,
    `productline`,
    COUNT(CASE WHEN `dealsize` = 'Small' THEN 1 END) AS `small_deals`,
    COUNT(CASE WHEN `dealsize` = 'Medium' THEN 1 END) AS `medium_deals`,
    COUNT(CASE WHEN `dealsize` = 'Large' THEN 1 END) AS `large_deals`,
    SUM(`quantityordered`) AS `total_quantity_sold`,
    CONCAT('$', FORMAT(SUM(`sales`), 2)) AS `total_sales_in_usd`
FROM
    `automotive_co`.`automotive_order`
WHERE
    `country` = 'USA'
GROUP BY
    `city`,`productline`
ORDER BY 
    `total_quantity_sold` DESC
LIMIT
    30
;    

30 rows affected.


[('San Rafael', 'Classic Cars', 21, 39, 7, Decimal('2381'), '$282,079.44'),
 ('San Rafael', 'Vintage Cars', 23, 22, 4, Decimal('1753'), '$168,967.80'),
 ('New York', 'Classic Cars', 10, 27, 7, Decimal('1644'), '$190,137.20'),
 ('San Rafael', 'Trucks and Buses', 15, 21, 0, Decimal('1232'), '$124,434.52'),
 ('San Francisco', 'Vintage Cars', 12, 11, 2, Decimal('932'), '$89,209.26'),
 ('New Bedford', 'Classic Cars', 9, 11, 1, Decimal('705'), '$78,619.00'),
 ('Brickhaven', 'Classic Cars', 8, 10, 2, Decimal('701'), '$74,917.27'),
 ('Nashua', 'Classic Cars', 5, 12, 1, Decimal('671'), '$72,833.39'),
 ('New York City', 'Classic Cars', 9, 8, 2, Decimal('662'), '$70,666.54'),
 ('Philadelphia', 'Classic Cars', 7, 12, 1, Decimal('621'), '$73,849.74'),
 ('Las Vegas', 'Classic Cars', 11, 9, 0, Decimal('611'), '$62,231.86'),
 ('San Francisco', 'Motorcycles', 9, 8, 1, Decimal('583'), '$62,876.15'),
 ('San Jose', 'Vintage Cars', 5, 6, 2, Decimal('556'), '$51,279.16'),
 ('San Jose', 'Classic Cars', 3, 4, 3, Decimal('511'), '$49,869.29'),
 ('Nashua', 'Vintage Cars', 6, 6, 1, Decimal('493'), '$49,674.39'),
 ('New York City', 'Motorcycles', 6, 10, 0, Decimal('491'), '$57,036.04'),
 ('Pasadena', 'Motorcycles', 3, 9, 1, Decimal('468'), '$48,677.39'),
 ('San Jose', 'Planes', 8, 4, 1, Decimal('450'), '$46,492.34'),
 ('New York', 'Motorcycles', 10, 4, 1, Decimal('440'), '$41,261.64'),
 ('New York City', 'Ships', 8, 5, 0, Decimal('439'), '$39,640.76'),
 ('New York', 'Trucks and Buses', 6, 5, 1, Decimal('437'), '$46,109.85'),
 ('Cambridge', 'Motorcycles', 5, 7, 0, Decimal('429'), '$43,435.92'),
 ('Newark', 'Planes', 3, 4, 2, Decimal('418'), '$34,727.53'),
 ('Bridgewater', 'Classic Cars', 5, 5, 2, Decimal('417'), '$53,314.88'),
 ('Glendale', 'Planes', 5, 7, 0, Decimal('413'), '$37,283.68'),
 ('Brickhaven', 'Vintage Cars', 4, 6, 1, Decimal('380'), '$39,792.91'),
 ('Boston', 'Ships', 5, 6, 0, Decimal('375'), '$31,673.43'),
 ('San Rafael', 'Ships', 5, 5, 0, Decimal('371'), '$30,886.58'),
 ('Boston', 'Classic Cars', 5, 5, 1, Decimal('369'), '$40,259.03'),
 ('Brisbane', 'Vintage Cars', 6, 5, 0, Decimal('367'), '$34,100.03')]

- - this table provied the specifics for each city in the US market.

The initial observation highlights the dominance of "Classic Cars" in sales, reinforcing its position as a crucial product line for the company. Ensuring the sustained success and potential improvement of Classic Cars sales should be a top priority.

Moving to the analysis of deal sizes, the data suggests that larger deals are less favored, with medium-sized deals being the most favorable, closely followed by small deals. This insight indicates that dedicating fewer resources to larger deals could be a strategic move. However, it's essential to delve deeper into understanding why larger deals are less popular. There might be underlying factors related to presentation or other aspects that need exploration.

On the geographical front, the city of San Rafael stands out as a robust revenue generator, contributing significantly to the overall sales. Strengthening the company's presence in San Rafael is recommended, emphasizing strategic expansion initiatives. This includes investing in market penetration strategies, exploring potential partnerships, and contemplating the establishment of new outlets. Simultaneously, operational optimization efforts are crucial, involving the adoption of lean processes, favorable supplier negotiations, and streamlined logistics to enhance efficiency and reduce costs.


# 3. Finding the Season Patterns

To understand seasonal trends and patterns in the automotive order data, we'll analyze the `orderdate` column using the `DATE` syntax. This helps us spot any patterns or trends in the data, providing insights into seasonal variations that can impact decision-making.

In [77]:
%%sql

SELECT
    WEEK(`orderdate`) AS `wzek`,
    MONTH(`orderdate`) AS `month`,
    YEAR(`orderdate`) AS `year`,
    SUM(`quantityordered`) AS `total_quantity_sold`,
    CONCAT('$', FORMAT(SUM(`sales`), 2)) AS `total_sales_in_usd`
FROM 
    automotive_co.automotive_order
GROUP BY 
    `week`,`month`,`year`
ORDER BY
    `total_sales_in_usd` DESC
LIMIT
    15
;

15 rows affected.


[(37, 9, 2023, Decimal('704'), '$97,279.19'),
 (46, 11, 2022, Decimal('1014'), '$96,869.23'),
 (6, 2, 2022, Decimal('993'), '$96,283.18'),
 (29, 7, 2023, Decimal('669'), '$95,921.24'),
 (30, 7, 2022, Decimal('955'), '$95,564.65'),
 (52, 12, 2023, Decimal('706'), '$92,064.82'),
 (42, 10, 2022, Decimal('914'), '$91,154.60'),
 (39, 10, 2022, Decimal('89'), '$9,044.95'),
 (18, 5, 2023, Decimal('680'), '$87,906.24'),
 (7, 2, 2023, Decimal('689'), '$87,793.68'),
 (49, 12, 2023, Decimal('827'), '$86,751.24'),
 (32, 8, 2023, Decimal('725'), '$86,181.01'),
 (38, 9, 2023, Decimal('769'), '$85,547.69'),
 (40, 10, 2023, Decimal('634'), '$84,214.38'),
 (48, 11, 2023, Decimal('548'), '$83,393.38')]


To enhance clarity, we'll have transform the orderdate into diffrent year and fiscal quarters, categorizing the months as follows:

- January, February, and March (Q1)
- April, May, and June (Q2)
- July, August, and September (Q3)
- October, November, and December (Q4)

This approach simplifies the representation of temporal patterns, facilitating a more straightforward analysis of quarterly trends.


In [124]:
%%sql

SELECT
    CONCAT('Q', QUARTER(`orderdate`)) AS 'quarterly_order',
    SUM(`quantityordered`) AS `total_quantity_sold`,
    CONCAT('$', FORMAT(SUM(`sales`), 2)) AS `total_sales_in_usd`
FROM 
    automotive_co.automotive_order
GROUP BY
    `quarterly_order`
ORDER BY
    `total_sales_in_usd` DESC
;

4 rows affected.


[('Q4', Decimal('25757'), '$2,613,275.74'),
 ('Q3', Decimal('24128'), '$2,482,771.32'),
 ('Q1', Decimal('24068'), '$2,472,408.20'),
 ('Q2', Decimal('25114'), '$2,464,173.59')]

The relatively minimal change in the value of sales across quarters implies that the specific quarter in which transactions occurred did not significantly influence the total sales. 

- -


One of the more popular time series analyses is Year-over-Year Growth. While this type of analysis is well-suited for visualization, it is still helpful to see the sales amount in a particular year and month alongside the sales from the previous year. This analysis relies on the DATE and LAG syntax to generate data that fit the required criteria.

In [150]:
%%sql

SELECT
    YEAR(`orderdate`) AS `year`,
    MONTH(`orderdate`) AS `month`,
    SUM(`sales`) AS `total_sales`,
    LAG(SUM(`sales`), 12) OVER (ORDER BY YEAR(`orderdate`), MONTH(`orderdate`)) AS `sales_last_year`,
    (SUM(`sales`) - LAG(SUM(`sales`), 12) OVER (ORDER BY YEAR(`orderdate`), MONTH(`orderdate`))) AS `year_over_year_difference`
FROM 
    automotive_co.automotive_order
GROUP BY
    `year`, `month`
ORDER BY
    `year`, `month`
;


37 rows affected.


[(2021, 1, Decimal('214517.02'), None, None),
 (2021, 2, Decimal('153934.73'), None, None),
 (2021, 3, Decimal('201756.32'), None, None),
 (2021, 4, Decimal('213859.00'), None, None),
 (2021, 5, Decimal('204710.17'), None, None),
 (2021, 6, Decimal('230952.66'), None, None),
 (2021, 7, Decimal('215522.86'), None, None),
 (2021, 8, Decimal('220516.32'), None, None),
 (2021, 9, Decimal('173657.81'), None, None),
 (2021, 10, Decimal('238122.77'), None, None),
 (2021, 11, Decimal('196809.46'), None, None),
 (2021, 12, Decimal('176791.07'), None, None),
 (2022, 1, Decimal('274131.12'), Decimal('214517.02'), Decimal('59614.10')),
 (2022, 2, Decimal('239686.37'), Decimal('153934.73'), Decimal('85751.64')),
 (2022, 3, Decimal('319724.15'), Decimal('201756.32'), Decimal('117967.83')),
 (2022, 4, Decimal('271183.18'), Decimal('213859.00'), Decimal('57324.18')),
 (2022, 5, Decimal('237554.32'), Decimal('204710.17'), Decimal('32844.15')),
 (2022, 6, Decimal('253186.13'), Decimal('230952.66'), Decimal('22233.47')),
 (2022, 7, Decimal('282628.43'), Decimal('215522.86'), Decimal('67105.57')),
 (2022, 8, Decimal('330217.50'), Decimal('220516.32'), Decimal('109701.18')),
 (2022, 9, Decimal('230425.85'), Decimal('173657.81'), Decimal('56768.04')),
 (2022, 10, Decimal('338011.70'), Decimal('238122.77'), Decimal('99888.93')),
 (2022, 11, Decimal('326496.27'), Decimal('196809.46'), Decimal('129686.81')),
 (2022, 12, Decimal('263900.00'), Decimal('176791.07'), Decimal('87108.93')),
 (2023, 1, Decimal('282491.78'), Decimal('274131.12'), Decimal('8360.66')),
 (2023, 2, Decimal('294153.15'), Decimal('239686.37'), Decimal('54466.78')),
 (2023, 3, Decimal('360813.47'), Decimal('319724.15'), Decimal('41089.32')),
 (2023, 4, Decimal('288268.49'), Decimal('271183.18'), Decimal('17085.31')),
 (2023, 5, Decimal('452223.19'), Decimal('237554.32'), Decimal('214668.87')),
 (2023, 6, Decimal('312236.45'), Decimal('253186.13'), Decimal('59050.32')),
 (2023, 7, Decimal('363136.54'), Decimal('282628.43'), Decimal('80508.11')),
 (2023, 8, Decimal('332697.08'), Decimal('330217.50'), Decimal('2479.58')),
 (2023, 9, Decimal('333968.93'), Decimal('230425.85'), Decimal('103543.08')),
 (2023, 10, Decimal('348863.52'), Decimal('338011.70'), Decimal('10851.82')),
 (2023, 11, Decimal('356987.49'), Decimal('326496.27'), Decimal('30491.22')),
 (2023, 12, Decimal('367293.46'), Decimal('263900.00'), Decimal('103393.46')),
 (2024, 1, Decimal('131200.09'), Decimal('282491.78'), Decimal('-151291.69'))]

The numbers in the `year_over_year_difference` column consistently show positive values, which means sales have been going up month by month and year by year. However, when you look at the monthly differences between consecutive years, those numbers seem a bit all over the place without any clear pattern. Even though we're seeing a positive growth trend annually, it's important to mention that this might not be statistically significant. For a more in-depth analysis, we'll need to dig into some statistical methods outside of MySQL.

# 4. Analyzing MSRP Margin's Impact on Sales


The most fitting way to see how the margin of MSRP and selling prices are related with sales through Pearson correlation analysis. Normally, this analysis would be short and sweet, but because MySQL doesn't have a handy `CORR` function like some other databases, we have to use a few tricks to get that correlation value.


The mathematical representation of this query is as follows

`correlation = Σᵢ₌₁ⁿ [(stddev_sales * (salesᵢ - avg_sales) / stddev_sales) * ((stddev_price_margin * ((msrpᵢ - price_eachᵢ) - avg_price_margin) / stddev_price_margin)] / (n - 1)`



In [156]:
%%sql

SELECT
    SUM((sales - avg_sales) / stddev_sales * ((msrp - price_each) - avg_price_margin) / stddev_price_margin) 
    / (COUNT(*) - 1) AS correlation
FROM automotive_co.automotive_order
CROSS JOIN (
    SELECT
        AVG(sales) AS avg_sales,
        STDDEV(sales) AS stddev_sales,
        AVG(msrp - price_each) AS avg_price_margin,
        STDDEV(msrp - price_each) AS stddev_price_margin
    FROM automotive_co.automotive_order
) AS normalization_stats
WHERE (msrp - price_each) IS NOT NULL
;


1 rows affected.


[(0.40108281500895393,)]

The value of `0.40108281500895393` suggest that there is a moderate positive correlation between MSRP (Manufacturer's Suggested Retail Price) and actual sales. Although this values shows us that as the MSRP Margin tend to increase and so does sales, however this relationship is moderate and not extremely strong. 

To avoid falling into the correlation is causation trap, we need to do more analysis to better precive this relationship. But, because the limitation of this data there seems to only be a limited number of analysis that we can do, one of them being time series analysis.

In [162]:
%%sql

WITH yearly_data AS (
    SELECT
        EXTRACT(YEAR FROM orderdate) AS year,
        SUM(sales) AS total_sales,
        AVG(sales) AS avg_sales,
        STDDEV(sales) AS stddev_sales,
        AVG(msrp - price_each) AS avg_price_margin,
        STDDEV(msrp - price_each) AS stddev_price_margin
    FROM automotive_co.automotive_order
    WHERE (msrp - price_each) IS NOT NULL
    GROUP BY year
)

SELECT
    year,
    AVG((sales - avg_sales) / stddev_sales * (msrp - price_each - avg_price_margin) / stddev_price_margin) AS correlation
FROM 
    automotive_co.automotive_order
CROSS JOIN 
    yearly_data
WHERE 
    EXTRACT(YEAR FROM orderdate) = yearly_data.year
GROUP BY 
    year
ORDER BY 
    year;


4 rows affected.


[(2021, -0.1866561516372124),
 (2022, 0.3272255057653009),
 (2023, 0.42128118404247705),
 (2024, 0.6360282402371316)]

In [169]:
%%sql
SELECT
    MONTH(`orderdate`) AS `month`,
    AVG(`msrp`) AS `avg_msrp`,
    AVG(`price_each`) AS `avg_price`,
    AVG(`msrp` - `price_each`) AS `avg_msrp_margin`,
    CONCAT('$', FORMAT(SUM(`sales`), 2)) AS `total_sales_in_usd`
FROM
    automotive_co.automotive_order
WHERE
    YEAR(`orderdate`) = '2021'
GROUP BY
    `month`
ORDER BY
    `total_sales_in_usd` DESC
;

    

12 rows affected.


[(2, Decimal('81.6735'), Decimal('81.710204'), Decimal('-0.036735'), '$153,934.73'),
 (9, Decimal('81.4211'), Decimal('80.989825'), Decimal('0.431228'), '$173,657.81'),
 (12, Decimal('77.0000'), Decimal('74.798971'), Decimal('2.201029'), '$176,791.07'),
 (11, Decimal('77.3836'), Decimal('78.006575'), Decimal('-0.623014'), '$196,809.46'),
 (3, Decimal('81.2647'), Decimal('81.737353'), Decimal('-0.472647'), '$201,756.32'),
 (5, Decimal('76.1250'), Decimal('77.125694'), Decimal('-1.000694'), '$204,710.17'),
 (4, Decimal('78.5132'), Decimal('76.999868'), Decimal('1.513289'), '$213,859.00'),
 (1, Decimal('78.2029'), Decimal('79.454783'), Decimal('-1.251884'), '$214,517.02'),
 (7, Decimal('79.6324'), Decimal('78.635882'), Decimal('0.996471'), '$215,522.86'),
 (8, Decimal('78.2877'), Decimal('78.153562'), Decimal('0.134110'), '$220,516.32'),
 (6, Decimal('79.7821'), Decimal('77.277821'), Decimal('2.504231'), '$230,952.66'),
 (10, Decimal('85.7917'), Decimal('84.463056'), Decimal('1.328611'), '$238,122.77')]

Across the years analysis of the correlation between adjustments in the MSRP Margin and sales reveals interesting trends. In `2021`, a marginal negative association of approximately `-0.19` suggests a limited impact on sales. However, the subsequent years portray a positive correlation, with `2022` showing a noteworthy coefficient of around `0.33`, and `2023` exhibiting a moderate correlation of approximately `0.42`. The trend peaks in 2024, where a robust positive correlation of about `0.64` indicates a substantial connection between elevating the MSRP Price Margin and notable increases in sales.


`he correlation between MSRP Margin and sales appears to vary`, showing a moderate strength in some years and a weaker association in others. This outcome suggests that while there is a definite impact of MSRP Margin on sales, `the influence might not be consistently strong across different periods`. It implies that `selling products below the MSRP price could positively impact sales, but other factors likely play a significant role in driving sales increases`.
